shelf/utils.py

```diff
@@ -20,6 +21,23 @@ def create_path(*args):
     return full_path
 
 
+def get_host_from_uri(uri):
+    """
+        Gets the host (which is actually the protocol + the
+        hostname) from a full URI.
+
+        Args:
+            uri(string)
+
+        Returns:
+            string
+    """
+    parts = urlparse.urlparse(uri)
+    host = parts.scheme + "://" + parts.hostname
```

Comment:
Right I chatted to you about it and remembered it had to do with being behind a load balancer. Although this is not a strategic way to fix this problem, I am going forward with the "cutting corners" directive we've been given.

I've added a comment.